ecs-init/cache/cache.go

```diff
@@ -25,24 +25,30 @@ import (
 	"strings"
 
 	"github.com/aws/amazon-ecs-init/ecs-init/config"
+	"github.com/aws/aws-sdk-go/aws/ec2metadata"
+	"github.com/aws/aws-sdk-go/aws/session"
 	log "github.com/cihub/seelog"
 )
 
 const (
 	orwPerm = 0700
 )
 
-// Downloader is resposible for cache operations relating to downloading the agent
 type Downloader struct {
-	getter httpGetter
-	fs     fileSystem
+	getter   httpGetter
+	fs       fileSystem
+	metadata instanceMetadata
+
+	region string
 }
 
 // NewDownloader returns a Downloader with default dependencies
 func NewDownloader() *Downloader {
+	sessionInstance := session.Must(session.NewSession())
```

Comment:
`session.Must` will panic if it can't create the session.  Use `session.NewSession` and check for an error.  If there is an error, we can fall-back to not querying `ec2metadata` and just use the default bucket.